infobind/cache.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infobind
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/terror"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/sqlexec"
+	log "github.com/sirupsen/logrus"
+)
+
+const defaultBindCacheSize = 5
```

Comment:
Why do we need this?

Reply:
bindCache is a map, so add a constant in here

Reply:
How about making it as a config item? You can do it in another PR if it is acceptable.

Reply:
I think delete this const also is ok. This config only is used in initialize the first bind cache when tidb server startup.

Reply:
So the bind cache size may grow larger than `defaultBindCacheSize`?